zerver/actions/streams.py

```diff
@@ -1309,7 +1309,7 @@ def do_change_stream_permission(
             stream_ids=[stream.id],
             user_ids=sorted(stream_subscriber_user_ids),
         )
-        send_event(stream.realm, peer_add_event, peer_notify_user_ids)
+        send_event_on_commit(stream.realm, peer_add_event, peer_notify_user_ids)
```

Comment:
It might be cleaner to do these secondary files as prep commits, and try to do them somewhat completely -- e.g. a commit sweeping `zerver/actions/uploads.py` for using transactions consistently as well as using the `send_event_on_commit` variants.

Reply:
We [discussed it yesterday on CZO](https://chat.zulip.org/#narrow/stream/9-issues/topic/jump.20to.20topic.20view.20before.20message.20is.20there/near/1800498)